packages/keyv/src/index.js

```diff
@@ -73,7 +70,7 @@ class Keyv extends EventEmitter {
 		// Attach iterators
 		if (typeof this.opts.store[Symbol.iterator] === 'function' && this.opts.store instanceof Map) {
 			this.iterator = generateIterator(this.opts.store);
-		} else if (typeof this.opts.store.iterator === 'function') {
+		} else if (typeof this.opts.store.iterator === 'function' && this.opts.store.opts?.dialect === 'sqlite') { // Iterator only supported for sqlite for now
```

Comment:
@jaredwray added this since I only added for SQLITE for now. I wll add for remaining ones